packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -173,6 +177,25 @@ impl Abigen {
                     impl #methods_name {
                         #contract_functions
                     }
+
+
+                    impl Logging for #name
+                    {
+                        fn get_logs(receipts: &[Receipt]) -> Vec<String> {
+                            #name::get_logs(&receipts)
+                        }
+
+                        fn logs_with_type<D: Tokenizable + Parameterize>(receipts: &[Receipt]) -> Result<Vec<D>, SDKError> {
+                            Self::logs_with_type::<D>(receipts)
+                        }
+                    }
+
+                    // Implemented manually so that we can use Debug on types that use depend on the Logging trait
+                    impl fmt::Debug for #name {
+                        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                            write!(f, "{}", #name_str)
```

Comment:
Any reason this differs from the derived version of Debug? I.e. why not `derive(Debug)` it?

Reply:
I thought we can not use the macro inside of our procedural macro. I was wrong :D Thanks!